src/submission/models.py

```diff
@@ -2323,6 +2372,7 @@ def get_or_snapshot_if_email_found(cls, email, article):
         except cls.DoesNotExist:
             try:
                 account = core_models.Account.objects.get(email=email)
+                account.add_account_role('author', article.journal)
                 author = account.snapshot_as_author(article)
```

Comment:
seeing the following line, I wonder if we could delegate this to `Account.snapshot_as_author` method instead, so we can avoid requiring this step elsewhere